app/redux/thunks.js

```diff
@@ -0,0 +1,154 @@
+import axios from 'axios';
+
+import {
+    setProducts,
+    setSingleProduct,
+    setUser,
+    setOrders,
+} from 'actions.js';
+
+//Thunk for fetching all products. 
+//Do you think we could initialize id to null, and reuse this thunk to make a call for a single product?
+export const fetchProducts = () => {
+    return dispatch => {
+        return axios.get(`/api/products`)
+        .then(res => dispatch(setProducts(res.data)))
+        .catch(e => console.error('Error fetching products', e))
+    }
+}
+
+//Thunk for fetching a single product
+export const fetchSingleProduct = productId => {
+    return dispatch => {
+        return axios.get(`/api/products/${ productId }`)
+        .then(res => dispatch(setSingleProduct(res.data)))
+        .catch(e => console.error('Error fetching single product', e))
+    }
+}
+
+//Thunk for creating a product.
+//Refetches products after creating.
+export const postProduct = product => {
+    return dispatch => {
+        return axios.post(`/api/products`, product)
+        .then(() => dispatch(fetchProducts()))
```

Comment:
My redux isn't great, but is this meant to dispatch a function that dispatches? Can you just use .then(() => fetchProducts()) ?

Reply:
Make sense. Will try it.